lib/launch.c

```diff
@@ -177,14 +189,42 @@ launch_at_stage3(const char *name, lcore_function_t *f, void *arg,
 	else
 		list_add(&entry->list, &launch_heads.stage3);
 
+	lcores_info.num_lcores++;
+
 	return 0;
 
-name_cpy:
-	rte_free(name_cpy);
+name_cpy2:
+	rte_free(name_cpy2);
+name_cpy1:
+	rte_free(name_cpy1);
 fail:
 	return -1;
 }
 
+uint16_t
+get_filtered_num_lcores(const char *block_names[], uint8_t num_blocks)
```

Comment:
1. Rename `get_filtered_num_lcores()` to`launch_count_lcores()`. This new name emphasize the name of the library.

2. Rename parameter `block_names` to `filtered_blocks`.

3. Let's make `launch_count_lcores()` future proof changing its return type from `uint16_t` to `int`.